### ISSUE TITLE
attention_lstm.py and Tensorflow

### ISSUE THREAD

**Issue by jsuit** [philipperemy/keras-attention-mechanism#6]
In the attention_3d_block, I have some questions/bug (I think). I am running on Tensorflow. 
(1) inputs doesn't have a shape method. So it crashes. I assume you meant to call the shape function on  the numpy array on inputs_1.
(2) Is there a reason for calling Permute?
(3) What is the Reshape layer supposed to do? After the call to Permute, isn't the output of the previous permute layer already in shape (Batch Size, input_dim, TIME_STEPS)?
(4) The next call to Dense expects ndim =2, not 3. So the code crashes for me. I assume you meant the previous Reshape layer to map the 3d input to 2d?
(5) I would just like to point out that APPLY_ATTENTION_BEFORE_LSTM is False iff you call model_attention_applied_before_lstm.

**Comment by philipperemy** [comment]
1/ Make sure you use the latest versions Keras 2.0.2 and Tensorfow 1.2.0
2/ Yes when you apply a Dense layer, it applies to the last dimension of your tensor. I use permute to apply a Dense layer along the time axis (by default it's axis=1 in Keras).
3/ Reshape has zero purpose except maybe making the code more explicit and clear.
4/ Make sure you use the latest versions Keras 2.0.2 and Tensorfow 1.2.0
5/ Yes APPLY_ATTENTION_BEFORE_LSTM is a flag to apply LSTM either before or after.
